js/src/wp-seo-replacevar-plugin.js

```diff
@@ -1,70 +1,327 @@
-/* global wpseoReplaceVarsL10n */
+/* global wpseoReplaceVarsL10n, require, YoastSEO */
 (function() {
 	'use strict';
 
+	var forEach = require( 'lodash/forEach' );
+	var filter = require( 'lodash/filter' );
+	var isUndefined = require( 'lodash/isUndefined' );
+	var ReplaceVar = require( './values/replaceVar' );
+
+	var modifiableFields = [
+		'content',
+		'title',
+		'snippet_title',
+		'snippet_meta',
+		'primary_category',
+		'data_page_title',
+		'data_meta_desc'
+	];
+
+	var placeholders = {};
+	var taxonomyElements = {};
+
 	/**
-	 * variable replacement plugin for wordpress.
+	 * Variable replacement plugin for WordPress.
 	 */
 	var YoastReplaceVarPlugin = function( app ) {
 		this._app = app;
-
-		this.replaceVars = wpseoReplaceVarsL10n.replace_vars;
-
 		this._app.registerPlugin( 'replaceVariablePlugin', { status: 'ready' } );
 
+		this.registerReplacements();
 		this.registerModifications();
+		this.registerEvents();
+	};
+
+	YoastReplaceVarPlugin.prototype.registerReplacements = function() {
+		this.addReplacement( new ReplaceVar( '%%currentdate%%',     'currentdate' ) );
+		this.addReplacement( new ReplaceVar( '%%currentday%%',      'currentday' ) );
+		this.addReplacement( new ReplaceVar( '%%currentmonth%%',    'currentmonth' ) );
+		this.addReplacement( new ReplaceVar( '%%currenttime%%',     'currenttime' ) );
+		this.addReplacement( new ReplaceVar( '%%currentyear%%',     'currentyear' ) );
+		this.addReplacement( new ReplaceVar( '%%date%%',            'date' ) );
+		this.addReplacement( new ReplaceVar( '%%id%%',              'id' ) );
+		this.addReplacement( new ReplaceVar( '%%page%%',            'page' ) );
+		this.addReplacement( new ReplaceVar( '%%searchphrase%%',    'searchphrase' ) );
+		this.addReplacement( new ReplaceVar( '%%sitedesc%%',        'sitedesc' ) );
+		this.addReplacement( new ReplaceVar( '%%sitename%%',        'sitename' ) );
+		this.addReplacement( new ReplaceVar( '%%category%%',        'category' ) );
+
+		this.addReplacement( new ReplaceVar( '%%sep%%(\s+%%sep%%)*', 'sep' ) );
```

Comment:
Make sure this is replaced as the last replacement. Because if a replacevariable is empty, we still only want one separator.